Tests/Functional/Asynchronous/main.swift

```diff
@@ -0,0 +1,184 @@
+// RUN: %{swiftc} %s -o %{built_tests_dir}/Asynchronous
```

Comment:
Just a few thoughts on this file. I don't think anything needs to be changed now for this PR, but I've been thinking a lot about these things especially in light of the discussion on https://github.com/apple/swift-corelibs-xctest/pull/61 and I want to make sure we keep iterating towards the most effective testing strategy we can manage to produce.

Overall I really like these tests! I like the way you've named the individual test cases and I like how straightforward the test method bodies are throughout. Nonetheless, when I first started reviewing this file I found it pretty opaque and difficult to navigate. I think this is because the expectations are mostly encoded in the `CHECK` lines at the top of the file, relatively far way from the code actually being tested. Additionally, the expected failures/passes are buried in a screenful of text with no syntax highlighting (at least in Xcode), so the signal-to-noise ratio is pretty poor when trying to parse the expectations.

This test file also presents several barriers to adding new tests in the future, should that prove necessary. Besides adding the new test method body and/or `XCTestCase` subclass, it is also necessary to figure out precisely where the new `CHECK` lines fit into the existing output. Once that is done, other numbers like the total number of tests executed and the line numbers of existing failure messages may need to be adjusted! :fearful: 

Some of this could be improved by reducing the specificity of the `CHECK` contents (e.g. wildcards for the total number of tests being run, wildcard line numbers for failures where we aren't specifically testing the file/line output, wildcard for a greater portion of the file path) but I don't know if there's any reasonable way to improve the proximity of the tests and their expected outcomes without a more radical change in approach.